app/presenters/signup_presenter.rb

```diff
@@ -23,29 +25,34 @@ def beta?
   end
 
   def choices?
-    multiple_facet_choice_data.present? || single_facet_choice_data[0]["facet_choices"].present?
+    multiple_facet_choice_data.present? || single_facet_choice_data[0]['facet_choices'].present?
   end
 
   def choices
-    multiple_facet_choice_data || single_facet_choice_data
+    return multiple_facet_choice_data unless multiple_facet_choice_data.empty?
+
+    single_facet_choice_data
   end
 
   def choices_formatted
-    choices.map do |choice|
-      {
-        label: choice['facet_name'],
-        value: choice['facet_id'],
-        checked: choice['prechecked'],
-        items: choice['facet_choices'].map do |facet_choice|
-          {
-            name: "filter[#{choice['facet_id']}][]",
-            label: facet_choice['radio_button_name'],
-            value: facet_choice['key'],
-            checked: facet_choice['prechecked'] || selected_choices.fetch(choice['facet_id'], []).include?(facet_choice['key'])
-          }
-        end
-      }
+    formatted_choices = choices.map do |choice|
+      if choice.dig('facet_choices')
```

Comment:
You'll get unhandled KeyError errors `'facet_choices'` is not part of the hash. Could do

```
next unless choice['facet_choices'].blank?
```